docs/DeveloperGuide.md

```diff
@@ -115,12 +115,41 @@ The `JsonAdaptedCompany` also contains a list of roles in `List<JsonAdaptedRole>
 ---
 ## **Implementation**
 
+### Find feature
+
+The `find` feature allows users to filter the company list by specifying company name keywords and role name keywords.
+
+#### Implementation
+The `find` command is primarily implemented by `FindCommandParser` a class that extends `Parser`, and `FindCommand`, which is a class that extends `Command`. For each `find` command, a `Predicate<Company>` object and a `Predicate<Role>` object will be created. Both `Predicate` objects contain a `test` function to determine whether the given company or role matches the keywords provided in the user input. 
+
+* Upon a valid user's input using the `find` command, the `FindCommandParser#parse()`creates a `CompanyNameContainsKeywordsPredicate` which extends `Predicate<Company>`, and a `RoleNameContainsKeywordsPredicate` which extends `Predicate<Role>`.
+* The `FindCommandParser#parse()` then uses the `CompanyNameContainsKeywordsPredicate` object and the `RoleNameContainsKeywordsPredicate` object to instantiate the `FindCommand`.
+* Then invoking the `FindCommand#execute()` method will update the `model` using the `Model#updateFilteredRoleList()` method, displaying only companies and roles that match the user input.
+
+
+![UML diagram of the Find feature](images/FindDiagram.png)
+
+The following sequence diagram shows how the `find` command operation works with the user input `find c/meta r/software mobile`:
+
+![UML diagram of the Find feature](images/FindSequenceDiagram.png)
+
+
+1. The user will first enter the input `find c/meta r/software mobile`, the `CompayListParser#parseCommand()` method will parse the information `c/meta r/software mobile` to `FindCommandParser` using the method `parse()` based on the keyword `find`.
+2. The `FindCommandParser#parse()` method will create a `CompanyNameContainsKeywordsPredicate` object with the company name keywords specified after the prefix `c/` and the role name keywords specified after the prefix `r/`. The `RoleNameContainsKeywordsPredicate` is also created using the role name keywords specified after the prefix `r/`.
```

Comment:
Not too sure if necessary, but perhaps you can clarify here that if a prefix is present, the keywords that follow must also exist?

Reply:
Thanks for pointing this out. Added it in.